apps/flame-defi/app/pool/modules/remove-liquidity/sections/content-section.tsx

```diff
@@ -6,15 +6,70 @@ import { RemoveAmountSlider } from "../components";
 import {
   usePoolContext,
   usePoolPositionContext,
-  useRemoveLiquidity,
+  useRemoveLiquidityPercentage,
+  useRemoveLiquidityTxn,
 } from "pool/hooks";
 import { ConfirmationModal } from "components/confirmation-modal/confirmation-modal";
 import { PoolTxnSteps } from "pool/components";
+import { TXN_STATUS } from "@repo/flame-types";
+import { useCallback } from "react";
 
 export const ContentSection = () => {
-  const { modalOpen, setModalOpen, txnStatus } = usePoolContext();
-  const { collectAsNative, handleCollectAsNative } = usePoolPositionContext();
-  const { liquidityToRemove, handlePercentToRemove } = useRemoveLiquidity();
+  const { modalOpen, setModalOpen } = usePoolContext();
+  const {
+    poolToken0,
+    poolToken1,
+    collectAsWrappedNative,
+    handleCollectAsWrappedNative,
+    refreshPoolPosition,
+  } = usePoolPositionContext();
+  const {
+    liquidityToRemove,
+    handlePercentToRemove,
+    percentageToRemove,
+    refreshLiquidityToRemove,
+  } = useRemoveLiquidityPercentage();
+  const {
+    removeLiquidity,
+    txnHash,
+    txnStatus,
+    setTxnStatus,
+    errorText,
+    setErrorText,
+  } = useRemoveLiquidityTxn(
+    liquidityToRemove,
+    collectAsWrappedNative,
+    percentageToRemove,
+  );
+  const isNativeToken =
```

Comment:
Please pick a more descriptive name. Seeing this in the markup first, I wondered "which token does this apply to?" And then I notice it also doesn't accurately represent whether native OR wrapped are in the pool.

`poolContainsNativeOrWrappedToken`